extensions/wikia/LinkSuggest/LinkSuggestLoader.class.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @class LinkSuggestLoader
+ * Proper loader for LinkSuggest
+ * @author TK-999
+ */
+class LinkSuggestLoader {
+	/**
+	 * @var array $selectors : Array of jQuery-style selectors to apply LinkSuggest to
+	 */
+	private $selectors = [];
+
+	/**
+	 * @var bool $userWantsLinkSuggest : Whether the current user has enabled linkSuggest
+	 */
+	private $userWantsLinkSuggest = false;
+
+	private static $instance = null;
+
+	private function __construct() {
+		$app = F::app();
+		$this->userWantsLinkSuggest = !( $app->wg->User->getGlobalPreference( 'disablelinksuggest' ) );
+
+		// only register hook if the user has enabled LinkSuggest
+		if ( $this->userWantsLinkSuggest ) {
+			$app->registerHook( 'BeforePageDisplay', get_class( $this ), 'onBeforePageDisplay', [], false, $this );
+		}
+	}
+
+	public static function getInstance() {
+		if ( self::$instance == null ) {
+			self::$instance = new LinkSuggestLoader();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Add the selector for an element to receive LinkSuggest
+	 * @param mixed $selectors : a single jQuery-style selector or an array
+	 */
+	public function addSelectors( $selectors ) {
+		// only add selectors if the user has enabled LinkSuggest
+		if ( $this->userWantsLinkSuggest ) {
+			if ( is_array( $selectors ) ) {
```

Comment:
I don't see any usage of this with array passed, do we need this implemented now?
Reason:
When something brakes, we'll have to fix this usecase even if it is not used anywhere.